### PR TITLE
CI: improve flacky test reporting job execution time

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Introduce caching step for the flaky tests reporting tool build artifacts, which should decrease avg. job execution time.

### Testing

- All CI checks shall pass
- The job in this PR checks shall skip Gutenberg build steps and run faster

Compare the following runs:
- https://github.com/woocommerce/woocommerce/actions/runs/9990209550/job/27611127491?pr=49665 (does complete build, takes 4:37)
- https://github.com/woocommerce/woocommerce/actions/runs/10002741468/job/27648896639?pr=49665 (reuses previously built artifact, takes 0:53)

### CODE REVIEW

File: plugins/woocommerce-admin/composer.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "woocommerce/woocommerce-admin",
-	"description": "WooCommerce admin.",
+	"description": "WooCommerce admin...",
```
Comment (kalessil) [inline plugins/woocommerce-admin/composer.json]: To be reverted before the merge.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -358,13 +353,44 @@ jobs:
...
+        with:
+          repository: WordPress/gutenberg
+          ref: dbf201449e9736f672b61e422787d47659db327a
+
+      - name: 'Pick pre-built @wordpress/report-flaky-tests resources'
+        if: ${{ !!env.FLAKY_REPORTS }}
+        uses: 'actions/cache@ab5e6d0c87105b4c9c2047343972218f562e4319'
+        with:
+          path: ./packages/report-flaky-tests/build/
+          key: "${{ runner.os }}-report-flaky-tests-TESTHASH"
```
Comment (kalessil) [inline .github/workflows/ci.yml]: TODO: update hash

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -358,13 +353,44 @@ jobs:
           # any output means there are reports
           echo "FLAKY_REPORTS=$(ls -A $downloadPath | head -1)" >> $GITHUB_ENV
 
-      - name: 'Setup'
+      - name: 'Checkout Gutenberg repository'
```
Comment (kalessil) [inline .github/workflows/ci.yml]: TODO: swap back - overrides merged reports

**Comment by adimoldovan** [comment]
This is a great improvement @kalessil, thank you for it. 
We were discussing on addressing it differently, by manually building the package and pushing the `dist` in the monorepo. I'm not sure which option is better, what do you think?

**Comment by kalessil** [comment]
@adimoldovan: pushing to dist sounds good, but I'm not sure how much additional work it will be - the build process in the Gutenberg repository seems not to bundle all dependencies.

I would propose to treat this PR as a band-aid (but create low-prio project issue for the project). I feel like E2E/PHPUnit tests speedup might take quite a bit of time, and if I mistaken we'll do as you suggested and on project scope - it's the right way to go.

**Review by adimoldovan — APPROVED** [review]

File: plugins/woocommerce/composer.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "woocommerce/woocommerce",
-	"description": "An eCommerce toolkit that helps you sell anything. Beautifully.",
+	"description": "An eCommerce toolkit that helps you sell anything. Beautifully...",
```
Comment (tammullen) [inline plugins/woocommerce/composer.json]: Is this accidental?

Comment (kalessil) [reply plugins/woocommerce/composer.json]: Triggers some jobs - I'm trying to add a pre-build version of the reporter, as Adi mentioned earlier. I will clean up before merging.

Comment (kalessil) [reply plugins/woocommerce/composer.json]: Cleaned up

**Comment by kalessil** [comment]
@adimoldovan: updated the PR to use pre-build version of flaky tests reporter and it works - https://github.com/woocommerce/woocommerce/actions/runs/10042357354/job/27753699333?pr=49665. I'll go ahead and merge, if you spot any necessary changes, I'll create a followup PR.

**Comment by adimoldovan** [comment]
> @adimoldovan: updated the PR to use pre-build version of flaky tests reporter and it works - [woocommerce/woocommerce/actions/runs/10042357354/job/27753699333?pr=49665](https://github.com/woocommerce/woocommerce/actions/runs/10042357354/job/27753699333?pr=49665). I'll go ahead and merge, if you spot any necessary changes, I'll create a followup PR.

Thanks @kalessil, it looks good. Can you add instructions in the readme on how to update the dist? Also, it would be helpful to know what is the current version (sha).

**Pull request merged**